hibernate-core/src/main/java/org/hibernate/engine/jdbc/batch/internal/BatchBuilderImpl.java

```diff
@@ -61,9 +61,15 @@ public void setJdbcBatchSize(int size) {
 
 	@Override
 	public Batch buildBatch(BatchKey key, JdbcCoordinator jdbcCoordinator) {
-		LOG.tracef( "Building batch [size=%s]", size );
-		return size > 1
-				? new BatchingBatch( key, jdbcCoordinator, size )
+		Integer batchSize = jdbcCoordinator.getJdbcSessionOwner().getBatchSize();
+		if(batchSize != null) {
```

Comment:
Please update your IDE code style. This line should have a space between the `if` and the `(`.  We prefer spaces inside the parens as well, but it is not enforced.